app/src/main/java/com/github/sdpteam15/polyevents/fragments/HomeFragment.kt

```diff
@@ -25,10 +25,49 @@ class HomeFragment : Fragment() {
         setHasOptionsMenu(true)
     }
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
-                              savedInstanceState: Bundle?): View? {
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_home, container, false)
+        val fragmentView = inflater.inflate(R.layout.fragment_home, container, false)
+        val linearLayout = fragmentView.findViewById<LinearLayout>(R.id.id_upcoming_activities_list)
+
+        val activities = ActivitiesQueryHelper.getUpcomingActivities()
+
+        for (activity: Activity in activities) {
+            setupActivityTab(activity, linearLayout)
+        }
+
+        return fragmentView
     }
 
+    /**
+     * Setup the layout for an activity tab and add it to the layout provided
+     * @param activity : the activity to add as a tab
+     * @param incomingActivities : the linear layout where to add this new activity tab
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun setupActivityTab(activity: Activity, incomingActivities: LinearLayout) {
+        val activityTab = layoutInflater.inflate(R.layout.tab_activity, null)
+
+        val activityName = activityTab.findViewById<TextView>(R.id.id_activity_name_text)
+        activityName.text = activity.name
+
+        val activitySchedule = activityTab.findViewById<TextView>(R.id.id_activity_schedule_text)
+        activitySchedule.text =  getString(R.string.at_hour_text, activity.getTime())
+
+        val activityZone = activityTab.findViewById<TextView>(R.id.id_activity_zone)
+        activityZone.text = activity.zone
+
+        val activityDescription = activityTab.findViewById<TextView>(R.id.id_activity_description)
+        activityDescription.text = activity.description
```

Comment:
What's the point of storing the value, we could only use findViewById().text = the string value no ?

Reply:
You're absolutely right, I will change it before merging.